File: scripts/ui/ANIlinks.js
const ANItex = require("ui/ANItex");

module.exports = {
    animaliaLinks: null,

    createLinks(){
        this.animaliaLinks = [
            new Links.LinkEntry("animalia", "https://github.com/FlinTyX/Animalia/releases", ANItex.frog, Pal.heal)
        ];
    },

    addLink(link){
        this.animaliaLinks.push(link);

        this.setup();
    },

    setup(){
        if(!this.animaliaLinks){
            this.createLinks();
        }

        const all = this.animaliaLinks.concat(Links.links),
        array = Packages.java.lang.reflect.Array.newInstance(Packages.mindustry.ui.Links.LinkEntry, all.length);

        all.forEach((e, i) => array[i] = e);

        Reflect.set(Links, "links", array);
    }
}

File: src/animalia/ui/AniLinks.java
package animalia.ui;

import arc.struct.Seq;
import arc.util.Reflect;
import mindustry.gen.Icon;
import mindustry.graphics.Pal;
import mindustry.ui.Links;

public class AniLinks {
    public static Seq<Links.LinkEntry> links = Seq.with(
        new Links.LinkEntry("animalia", "https://github.com/FlinTyX/Animalia/releases", AniTex.frog, Pal.heal),
        new Links.LinkEntry("report", AniUI.reportDialog.URL(), Icon.warning, Pal.remove)
    );

    public static void load(){
        links.add(Links.getLinks());

        Reflect.set(Links.class, "links", links.toArray(Links.LinkEntry.class));
    }
}
